components/movementsensor/mpu6050/mpu6050_test.go

```diff
@@ -178,3 +179,129 @@ func TestSuccessfulInitializationAndClose(t *testing.T) {
 	test.That(t, err, test.ShouldBeNil)
 	test.That(t, closeWasCalled, test.ShouldBeTrue)
 }
+
+func setupDependencies(mockData []byte) (config.Component, registry.Dependencies) {
+	testBoardName := "board"
+	i2cName := "i2c"
+
+	cfg := config.Component{
+		Name:  "movementsensor",
+		Model: model,
+		Type:  movementsensor.SubtypeName,
+		ConvertedAttributes: &AttrConfig{
+			BoardName:              testBoardName,
+			I2cBus:                 i2cName,
+			UseAlternateI2CAddress: true,
+		},
+	}
+
+	i2cHandle := &inject.I2CHandle{}
+	i2cHandle.ReadBlockDataFunc = func(ctx context.Context, register byte, numBytes uint8) ([]byte, error) {
+		if register == defaultAddressRegister {
+			return []byte{expectedDefaultAddress}, nil
+		}
+		return mockData, nil
+	}
+	i2cHandle.WriteByteDataFunc = func(ctx context.Context, b1, b2 byte) error {
+		return nil
+	}
+	i2cHandle.CloseFunc = func() error { return nil }
+	mockBoard := &inject.Board{}
+	mockBoard.I2CByNameFunc = func(name string) (board.I2C, bool) {
+		i2c := &inject.I2C{}
+		i2c.OpenHandleFunc = func(addr byte) (board.I2CHandle, error) {
+			return i2cHandle, nil
+		}
+		return i2c, true
+	}
+	return cfg, registry.Dependencies{
+		resource.NameFromSubtype(board.Subtype, testBoardName): mockBoard,
+	}
+}
+
+//nolint:dupl
+func TestLinearAcceleration(t *testing.T) {
+	linearAccelMockData := make([]byte, 16)
+	// x-accel
+	linearAccelMockData[0] = 64
+	linearAccelMockData[1] = 0
+	expectedAccelX := 9810.0
+	// y-accel
+	linearAccelMockData[2] = 32
+	linearAccelMockData[3] = 0
+	expectedAccelY := 4905.0
+	// z-accel
+	linearAccelMockData[4] = 16
+	linearAccelMockData[5] = 0
```

Comment:
Why are there 2 values for each dimension?

Reply:
each dimension comes through as 16-bit BigEndian values